### PR TITLE
searchBar

### PR DESCRIPTION
**Pull request by bso-odoo**

### CODE REVIEW

File: addons/web/static/tests/_framework/mock_templates.hoot.js [hunk 1]
```diff
@@ -70,7 +70,10 @@ export function makeTemplateFactory(name, factory) {
 
             // Patch "getTemplates" to access local cache
             exports.getTemplate = function mockedGetTemplate(name) {
-                const rawTemplate = getTemplate(name) || this.rawTemplates[name];
+                const rawTemplate = getTemplate(name) || this?.rawTemplates[name];
+                if (!rawTemplate) {
+                    return rawTemplate;
```
Comment (romo-odoo) [inline addons/web/static/tests/_framework/mock_templates.hoot.js]: Can we use `return;` to be more clear ?

File: addons/website/static/tests/interactions/snippets/search_bar_results.test.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const { core, el } = await startInteractions(searchTemplate);
+    const formEl = el.querySelector("form");
+    const inputEl = formEl.querySelector("input[type=search]");
+    const resultEls = formEl.querySelectorAll("a:has(.o_search_result_item)");
+    resultEls[1].focus();
+    await press("down");
+    expect(document.activeElement).toBe(inputEl);
+});
+
+test("searchbar selects previouos result on cursor up", async () => {
```
Comment (romo-odoo) [inline addons/website/static/tests/interactions/snippets/search_bar_results.test.js]: `windgardium previouso`

File: addons/website/static/src/snippets/s_searchbar/search_bar_results.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                    "max-height": `${document.body.clientHeight - bcr.bottom - 16}px !important`,
+                    "min-width": this.autocompleteMinWidth,
+                };
+            },
+            "t-att-class": () => ({
+                "dropdown": true,
+                "show": true,
+                "dropup": this.isDropup,
+            }),
+            "t-att-data-bs-popper": () => this.isDropup ? "" : undefined, // ???
```
Comment (romo-odoo) [inline addons/website/static/src/snippets/s_searchbar/search_bar_results.js]: Should this comment stay ?

File: addons/website/static/src/snippets/s_searchbar/000.js [hunk 4]
```diff
@@ -1,127 +1,107 @@
...
+            this.inputEl.setAttribute("autocomplete", "off");
         }
 
         this.options = {
-            'displayImage': this.displayImage,
-            'displayDescription': this.displayDescription,
-            'displayExtraLink': this.displayExtraLink,
-            'displayDetail': this.displayDetail,
-            'allowFuzzy': this.allowFuzzy,
+            "displayImage": this.displayImage,
```
Comment (ged-odoo) [inline addons/website/static/src/snippets/s_searchbar/000.js]: i guess you can simply define these values here, it seems like you don't need `this.displayImage` and the others:
```js
const dataset = this.inputEl.dataset;
this.options = {
  "displayImage": dataset.displayImage,
  "displayDescription": dataset.displayDescription,
  "displayExtraLink": dataset.displayExtraLink,
  "displayDetail": dataset.displayDetail,
  "allowFuzzy": dataset.noFuzzy,
};
```

File: addons/website/static/src/snippets/s_searchbar/000.js [hunk 5]
```diff
@@ -1,127 +1,107 @@
...
 
-        if (this.$input.data('noFuzzy')) {
-            $("<input type='hidden' name='noFuzzy' value='true'/>").appendTo(this.$input);
+    start() {
+        if (this.inputEl.dataset.noFuzzy) {
+            const noFuzzyEl = document.createElement("input");
+            noFuzzyEl.setAttribute("type", "hidden");
+            noFuzzyEl.setAttribute("name", "noFuzzy");
+            noFuzzyEl.setAttribute("value", "true");
+            this.insert(noFuzzyEl, this.inputEl);
```
Comment (ged-odoo) [inline addons/website/static/src/snippets/s_searchbar/000.js:82]: it looks like you do not remove noFuzzyEl when interaction is destroyed?

File: addons/website/static/src/snippets/s_searchbar/000.js [hunk 6]
```diff
@@ -1,127 +1,103 @@
...
+            this.registerCleanup(() => { noFuzzyEl.remove(); });
```
Comment (ged-odoo) [inline addons/website/static/src/snippets/s_searchbar/000.js]: oups sorry, just noticed that the code for insert is doing that:
```js
    insert(el, locationEl = this.el, position = "beforeend") {
        locationEl.insertAdjacentElement(position, el);
        this.registerCleanup(() => el.remove());
    }
```
forgot about that

**Pull request merged**